test/scale/azdisk_scale_test.go

```diff
@@ -0,0 +1,185 @@
+					MountPathGenerate: "/mnt/test-",
+				},
+			}
+			volumes = append(volumes, volume)
+		}
+
+		pod := testtypes.PodDetails{
+			Cmd:       testutil.ConvertToPowershellorCmdCommandIfNecessary("while true; do echo $(date -u) >> /mnt/test-1/data; sleep 3600; done"),
+			Volumes:   testutil.NormalizeVolumes(volumes, []string{}, isMultiZone),
+			IsWindows: testconsts.IsWindowsCluster,
+		}
+		test := PodSchedulingWithPVScaleTest{}
+
+		test.CSIDriver = testDriver
+		test.Pod = pod
+		test.Replicas = 1000
+		test.StorageClassParameters = map[string]string{"skuName": "Premium_LRS", "maxShares": "1", "cachingmode": "None"}
+
+		test.Run(cs, ns, schedulerName)
+	})
+
+	ginkgo.It("Scale test scheduling and rescheduling multiple pod with a persistent volume request on failover.", func() {
+		testutil.SkipIfUsingInTreeVolumePlugin()
+		testutil.SkipIfNotUsingCSIDriverV2()
+
+		podCount := 1000
+
+		volume := testtypes.VolumeDetails{
+			FSType:    "ext3",
+			ClaimSize: "10Gi",
+			VolumeMount: testtypes.VolumeMountDetails{
+				NameGenerate:      "test-volume-",
+				MountPathGenerate: "/mnt/test-",
+			},
+			MountOptions: []string{
+				"barrier=1",
+				"acl",
+			},
+		}
+
+		pod := testtypes.PodDetails{
+			Cmd:       testutil.ConvertToPowershellorCmdCommandIfNecessary("while true; do echo $(date -u) >> /mnt/test-1/data; sleep 3600; done"),
+			Volumes:   testutil.NormalizeVolumes([]testtypes.VolumeDetails{volume}, []string{}, isMultiZone),
+			IsWindows: testconsts.IsWindowsCluster,
+			UseCMD:    false,
+		}
+
+		test := PodSchedulingOnFailoverScaleTest{}
+		test.CSIDriver = testDriver
+		test.Pod = pod
+		test.PodCount = podCount
+		test.StorageClassParameters = map[string]string{"skuName": "StandardSSD_LRS"}
+
+		test.Run(cs, ns, schedulerName)
+	})
+
+	ginkgo.It("Scale test scheduling and starting multiple pod with multiple persistent volume requests.", func() {
+		testutil.SkipIfUsingInTreeVolumePlugin()
+		testutil.SkipIfNotUsingCSIDriverV2()
+
+		volMountedToPod := 3
+
+		volumes := []testtypes.VolumeDetails{}
+		for j := 1; j <= volMountedToPod; j++ {
+			volume := testtypes.VolumeDetails{
+				FSType:    "ext3",
+				ClaimSize: "10Gi",
+				VolumeMount: testtypes.VolumeMountDetails{
+					NameGenerate:      "test-volume-",
+					MountPathGenerate: "/mnt/test-",
+				},
+			}
+			volumes = append(volumes, volume)
+		}
+
+		pod := testtypes.PodDetails{
+			Cmd:       testutil.ConvertToPowershellorCmdCommandIfNecessary("echo 'hello world' > /mnt/test-1/data && grep 'hello world' /mnt/test-1/data"),
+			Volumes:   testutil.NormalizeVolumes(volumes, []string{}, isMultiZone),
+			IsWindows: testconsts.IsWindowsCluster,
+		}
+
+		test := PodSchedulingWithPVScaleTest{}
+		test.CSIDriver = testDriver
+		test.Pod = pod
+		test.Replicas = 350
+
+		test.Run(cs, ns, schedulerName)
+	})
+
+	ginkgo.It("Scale test scheduling and starting multiple pods with multiple persistent volume requests with replicas and reschedule on deletion.", func() {
```

Comment:
maxshares is 1 in this test. This test will not be creating replicas. We need maxshares >1 for creating replicas.

Reply:
Please refer to the above comment.